plugin/src/main/java/org/autorefactor/refactoring/rules/SwitchRefactoring.java

```diff
@@ -385,4 +428,99 @@ private Variable extractVariableWithConstantValue(Expression firstOp, Expression
         }
         return null;
     }
+
+    @Override
+    public boolean visit(final SwitchStatement node) {
+        List<Pair<List<Statement>, List<Statement>>> switchStructure = getSwitchStructure(node);
+
+        for (int referenceIndex = 0; referenceIndex < switchStructure.size() - 1; referenceIndex++) {
+            for (int comparedIndex = referenceIndex + 1; comparedIndex < switchStructure.size(); comparedIndex++) {
+                Pair<List<Statement>, List<Statement>> referenceCase = switchStructure.get(referenceIndex);
+                Pair<List<Statement>, List<Statement>> comparedCase = switchStructure.get(comparedIndex);
+
+                if (!referenceCase.getSecond().isEmpty()
+                        && breaksControlFlow(referenceCase.getSecond().get(referenceCase.getSecond().size() - 1))
+                        && isTheSameCode(referenceCase.getSecond(), comparedCase.getSecond())) {
+                    List<Statement> precedingStatements = switchStructure.get(comparedIndex - 1).getSecond();
+
+                    if (breaksControlFlow(precedingStatements.get(precedingStatements.size() - 1))) {
+                        refactorByMergingCases(false, referenceCase, comparedCase);
+                        return DO_NOT_VISIT_SUBTREE;
```

Comment:
You are right code has been written like this from the start because initially I wanted to keep all the return (DO_NOT_)VISIT_SUBTREE inside the `visit()` methods.

However I have changed my mind on this subject. For example, in this PR, I can easily see it leads to code duplication which is error prone, it makes the code longer and harder to read.

In short there are more cons than pros, so it is time to revisit this practice.